backend/user_canister/src/canister.rs

```diff
@@ -203,26 +210,41 @@ impl Canister {
         let chunk_id = request.chunk_id;
 
         // Update file content
-        //TODO CONSIDER PARTIAL: UPLOAD IN DISORDER
-        //TODO maybe add a check to verify all chunks are uploaded before marking as uploaded
-        //FIXME: mmm dont like it much. packet order is not guaranteed
         match &file.content {
             FileContent::Uploaded { .. } => {
                 return;
             }
             FileContent::PartiallyUploaded {
                 num_chunks,
+                uploaded_chunks,
                 file_type,
                 owner_key,
                 shared_keys,
             } => {
-                if chunk_id == *num_chunks - 1 {
+                // Check if the chunk is already uploaded
+                if uploaded_chunks.contains(&chunk_id) {
+                    return;
```

Comment:
We should add a return type for `upload_file_continue` such as `UploadFileContinueResponse` and return error variants both for `ChunkAlreadyUploaded` and `NoSuchFile`.